install-linux-dnf and authenticate doc updates (#3822)
MicrosoftDocs/azure-docs-cli: docs-ref-conceptual/authenticate-azure-cli.md

Locally, you can sign in interactively through your browser with the [az login](/cli/azure/reference-index#az-login) command. When you write scripts, the recommended approach is
to use service principals. By granting just the appropriate permissions needed to a service principal, you can keep your automation secure.

> Azure CLI doesn't store any of your login information. Instead, Azure CLI generates and stores an [authentication refresh token](/azure/active-directory/develop/v1-id-and-access-tokens#refresh-tokens). For more information on refresh and session token configuration, see [Refresh and session token lifetime policy properties](/azure/active-directory/develop/configurable-token-lifetimes#refresh-and-session-token-lifetime-policy-properties).

Reviewer: We only use refresh token, not session token.

```suggestion
Azure CLI doesn't store any of your login information. Instead, Azure CLI generates and stores an [authentication refresh token](/azure/active-directory/develop/v1-id-and-access-tokens#refresh-tokens). For more information on refresh token configuration, see [Refresh and session token lifetime policy properties](/azure/active-directory/develop/configurable-token-lifetimes#refresh-and-session-token-lifetime-policy-properties).
```

> Azure CLI doesn't store any of your login information. Instead, Azure CLI generates and stores an [authentication refresh token](/azure/active-directory/develop/v1-id-and-access-tokens#refresh-tokens). For more information on refresh and session token configuration, see [Refresh and session token lifetime policy properties](/azure/active-directory/develop/configurable-token-lifetimes#refresh-and-session-token-lifetime-policy-properties).

Reviewer: Regarding the first sentence

> Azure CLI doesn't store any of your login information.

In #1017: 

```diff
- None of your private credential information is stored locally.
+ None of your sign-in information is stored by the CLI.
```

This changed the meaning of the sentence. "private credential information" is not equivalent to "sign-in information".

#2664 then changed "sign-in" to "login".

The fact is that refresh token is only generated for user accounts. For service principal, we do save the credential (client secret / password) of the service principal, making this statement invalid.

```suggestion
When signing in with a user account, Azure CLI generates and stores an [authentication refresh token](/azure/active-directory/develop/v1-id-and-access-tokens#refresh-tokens). For more information on refresh and session token configuration, see [Refresh and session token lifetime policy properties](/azure/active-directory/develop/configurable-token-lifetimes#refresh-and-session-token-lifetime-policy-properties).
```
